src/Autocomplete.tsx

```diff
@@ -0,0 +1,83 @@
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
+
+import React, { useState, useCallback, useMemo } from 'react';
+import { Person } from './types/Person';
+
+interface Props {
+  people: Person[]
+  delay?: number
+  onSelected: (value: Person | null) => void
+}
+
+export const Autocomplete: React.FC<Props> = ({
+  people,
+  delay,
+  onSelected,
+}) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), [delay]);
+
+  const filteredPeople = useMemo(() => {
+    const normalizedQuery = appliedQuery.toLowerCase().trim();
+
+    return people.filter(
+      person => person.name.toLowerCase().includes(normalizedQuery),
+    );
+  }, [appliedQuery, people]);
+
+  const handleSelect = (person: Person) => {
+    setQuery(person.name);
+    onSelected(person);
+    setIsDropdownOpen(false);
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+    onSelected(null);
+  };
+
+  return (
+    <>
```

Comment:
it seems like [fragment](https://react.dev/reference/react/Fragment) is redundant